pycbc/filter/matchedfilter.py

```diff
@@ -1831,16 +1831,55 @@ def followup_event_significance(ifo, data_reader, bank,
     onsource_end += coinc_threshold
 
     # Calculate how much time needed to calculate significance
-    trim_pad = (data_reader.trim_padding * data_reader.strain.delta_t)
-    bdur = int(lookback + 2.0 * trim_pad + length_in_time)
-    if bdur > data_reader.strain.duration * .75:
-        bdur = data_reader.strain.duration * .75
+    trim_pad = data_reader.trim_padding * data_reader.strain.delta_t
+    buffer_duration = lookback + 2 * trim_pad + length_in_time
+    buffer_samples = bank.round_up(int(buffer_duration * bank.sample_rate))
+    max_safe_buffer_samples = int(
+        0.9 * data_reader.strain.duration * bank.sample_rate
+    )
```

Comment:
Why is this 0.9? I thought the desired lookback was 150 seconds? This is 10% of the strain duration (so 25 seconds ... or 52 if doubled, which is still less than the desired lookback).

Reply:
Yes, the desired lookback *is* 150 s (set in the function's arguments), but this is not setting the lookback, it is setting the *max possible* lookback given the current strain buffer. The 0.9 is just a paranoia fudge factor to account for off-by-a-tiny-amount errors in my thinking.